### ISSUE TITLE
Use the new Framework API 

### ISSUE THREAD

**Issue by jonniebigodes** [storybook-community/addon-queryparams#5]
Opening this issue so we can address the upcoming changes on how addons are built/updated for the upcoming 7.0 release. If anyone is interested in making the changes required and continuing to support and maintain the addon, let us know so that we're aware and help/guide the community moving forward. To help with this process, we've built an abridged guide [here](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#70-addon-authors-changes), and should any questions or issues. Please reach out to us in the prerelease channel in our [Discord Server](https://discord.gg/storybook).

**Comment by flensrocker** [comment]
Any news on this? This is the only addon which blocks me from migrating to Storybook 7.

Thank you!

**Comment by flensrocker** [comment]
For those who are blocked from migrating to Storybook 7 because of this addon.

I just use this function as a decorator and it "works for me". Lucky me I don't know enough about storybook addons and for what the rest of the code is needed for... 😎

```typescript
import * as qs from "qs";

export const withQuery = (getStory: (ctx: unknown) => unknown, context: { parameters?: { query?: string | object } }) => {
  const { location } = document;
  const currentQuery = qs.parse(location.search, { ignoreQueryPrefix: true });
  if (context.parameters?.query) {
    const additionalQuery =
      typeof context.parameters.query === "string"
        ? qs.parse(context.parameters.query, { ignoreQueryPrefix: true })
        : context.parameters.query;
    const newLocation = new URL(document.location.href);
    newLocation.search = qs.stringify({ ...currentQuery, ...additionalQuery });
    history.replaceState({}, document.title, newLocation.toString());
  }
  return getStory(context);
};
```

**Comment by yannbf** [comment]
Closing this as done and released in v7.0.0!
